zproject/dev_settings.py

```diff
@@ -89,6 +89,38 @@
 # Set this True to send all hotspots in development
 ALWAYS_SEND_ALL_HOTSPOTS = False  # type: bool
 
+# Toggle faking ldap auth using fakeldap.
+# ZulipLDAPAuthBackend has not been added to AUTHENTICATION_BACKENDS
+# here, user has to add that to enable fake LDAP server.
+FAKE_LDAP_AUTH_ENABLED = True
+
+# Three modes are allowed.
+#   (A) If users' email addresses are in LDAP and used as username.
+#   (B) If LDAP only has usernames but email addresses are of the form
+#       username@example.com
+#   (C) If LDAP username are completely unrelated to email addresses,
+# Refer `prod_settings_template.py` for description of the modes.
+FAKE_LDAP_MODE = 'a'
+
+# By default 8 users are populated in the LDAP directory.
+FAKE_LDAP_EXTRA_USER = 0
+
+import ldap
+from django_auth_ldap.config import LDAPSearch, GroupOfNamesType
+
+if FAKE_LDAP_AUTH_ENABLED:
+    LDAP_APPEND_DOMAIN = None
+    AUTH_LDAP_USER_DN_TEMPLATE = 'uid=%(user)s,ou=users,dc=zulip,dc=com'
+
+    if FAKE_LDAP_MODE == 'a':
+        AUTH_LDAP_USER_SEARCH = LDAPSearch("ou=users,dc=zulip,dc=com",
+                                           ldap.SCOPE_SUBTREE, "(email=%(user)s)")
+    if FAKE_LDAP_MODE == 'b':
+        LDAP_APPEND_DOMAIN = 'zulip.com'
+
```

Comment:
Move this space above the `FAKE_LDAP_MODE == 'b'` block, if you really intend to have an if/else setup for just these second two cases.